addons/website/models/ir_http.py

```diff
@@ -114,8 +114,15 @@ def _rewrite_len(self, website_id: int) -> int:
         return len(rewrites)
 
     def _get_rewrites(self, website_id):
+        if request:
+            cache = request.__dict__.setdefault('_website_rewrites_cache', {})
+            if website_id not in cache:
+                domain = [('redirect_type', 'in', ('308', '404')), '|', ('website_id', '=', False), ('website_id', '=', website_id)]
+                cache[website_id] = {rewrite.url_from: rewrite for rewrite in self.env['website.rewrite'].sudo().search(domain)}
```

Comment:
do we want to keep the rewrite sudo in the cache? Cannot we `rewrite.url_from: rewrite.sudo(False)`?